src/test/java/com/snowflake/kafka/connector/internal/streaming/schemaevolution/iceberg/ParseIcebergColumnTreeTest.java

```diff
@@ -0,0 +1,236 @@
+            "OBJECT(offset INT, topic VARCHAR(16777216), partition"
+                + " INT, key VARCHAR(16777216), schema_id INT, key_schema_id"
+                + " INT, CreateTime LONG, LogAppendTime LONG,"
+                + " SnowflakeConnectorPushTime LONG, headers MAP(VARCHAR(16777216),"
+                + " VARCHAR(16777216)))"));
+  }
+
+  @ParameterizedTest
+  @MethodSource("parseFromJsonArguments")
+  void parseFromJsonRecordSchema(String jsonString, String expectedType) {
+    // given
+    SinkRecord record = createKafkaRecord(jsonString, false);
+    JsonNode recordNode = RecordService.convertToJson(record.valueSchema(), record.value(), true);
+    IcebergColumnJsonValuePair columnValuePair =
+        IcebergColumnJsonValuePair.from(recordNode.fields().next());
+    // when
+    IcebergColumnTree tree = new IcebergColumnTree(columnValuePair);
+    // then
+    Assertions.assertEquals(expectedType, tree.buildType());
+    Assertions.assertEquals("TESTCOLUMNNAME", tree.getColumnName());
+  }
+
+  static Stream<Arguments> parseFromJsonArguments() {
+    return Stream.of(
+        arguments("{\"testColumnName\" : 1 }", "LONG"),
+        arguments(
+            "{ \"testColumnName\": {" + "\"k1\" : 1," + "\"k2\" : 2" + "} " + "}",
+            "OBJECT(k1 LONG, k2 LONG)"),
+        arguments(
+            "{ \"testColumnName\": {"
+                + "\"k1\" : { \"nested_key1\" : 1},"
+                + "\"k2\" : { \"nested_key2\" : 2}"
+                + "}}",
+            "OBJECT(k1 OBJECT(nested_key1 LONG), k2 OBJECT(nested_key2 LONG))"),
+        arguments(
+            "{ \"testColumnName\": {"
+                + "\"vehicle1\" : { \"car\" : { \"brand\" : \"vw\" } },"
+                + "\"vehicle2\" : { \"car\" : { \"brand\" : \"toyota\" } }"
+                + "}}",
+            "OBJECT(vehicle2 OBJECT(car OBJECT(brand VARCHAR)), "
+                + "vehicle1 OBJECT(car OBJECT(brand VARCHAR)))"),
+        // <- todo lol with k1, k2 the order is natural, however it changes an order when I used
+        // vehicles - inspect it
+        arguments(
+            "{ \"testColumnName\": {"
+                + "\"k1\" : { \"car\" : { \"brand\" : \"vw\" } },"
+                + "\"k2\" : { \"car\" : { \"brand\" : \"toyota\" } }"
+                + "}}",
+            "OBJECT(k1 OBJECT(car OBJECT(brand VARCHAR)), k2"
+                + " OBJECT(car"
+                + " OBJECT(brand"
+                + " VARCHAR)))"),
+        arguments(
```

Comment:
we should add the descripton to the argument as it;s getting quite complex

Reply:
I don't know. Should I verbally describe Jsons that I parse here...